tests/console/machinery.pm

```diff
@@ -13,13 +13,28 @@
 use base "consoletest";
 use strict;
 use testapi;
+use utils;
+use version_utils;
 
 sub run {
     select_console 'root-console';
-    assert_script_run 'which OneClickInstallCLI || zypper -n in yast2-metapackage-handler', 200;
-    my $ret_check = '[ "$?" = "141" ] || [ "$?" = "0" ]';
-    assert_script_run 'yes | OneClickInstallCLI http://machinery-project.org/machinery.ymp ; ' . $ret_check, 200;
+
+    if (is_sle) {
+        assert_script_run 'source /etc/os-release';
+        if (is_sle '>=15') {
+            assert_script_run 'SUSEConnect -p PackageHub/${VERSION_ID}/${CPU}';
+        }
+        else {
+            assert_script_run 'SUSEConnect -r ' . get_required_var('SCC_REGCODE');
+            assert_script_run 'SUSEConnect -p sle-module-adv-systems-management/12/${CPU}';
+        }
+    }
```

Comment:
Please use the existing helper functions: `git grep suseconnect`

Reply:
As I already pointed out in a different ticket the existing suseconnect functions are hard to read and undocumented.

Reply:
well, you can change that

Reply:
I could do that but it is out of scope for this ticket as it would affect other existing tests as I would just rewrite everything.
Feel free to create a ticket.

Reply:
Also it seems that I need to run the commands here in a special way anyway because I need to workaround the bug about finding the PackageHub Pool Metatata and re-running the command (ignoring the exit code when it runs the first time).
I guess this wouldn't be possible with that library anyway.

Reply:
@asmorodskyi WDYT?